src/components/Roles/AddRoleDrawer.tsx

```diff
@@ -0,0 +1,176 @@
+    addRole.mutateAsync(data);
+  };
+
+  return (
+    <Drawer position="end" size="medium" open={isOpen}>
+      <DrawerHeader>
+        <DrawerHeaderTitle
+          action={
+            <Button
+              appearance="subtle"
+              aria-label="Close"
+              disabled={addRole.isLoading}
+              icon={<DismissRegular />}
+              onClick={() => setIsOpen(false)}
+            />
+          }
+        >
+          Add Role
+        </DrawerHeaderTitle>
+      </DrawerHeader>
+      <DrawerBody>
+        <FormProvider {...myForm}>
+          <form id="addRoleForm" onSubmit={myForm.handleSubmit(onSubmit)}>
+            <div className="requestFieldContainer">
+              <BACCombobox<NewRole>
+                name="Title"
+                labelText="Role"
+                rules={{
+                  required: "Role is required",
+                }}
+              >
+                {ROLES.map((role) => (
+                  <Option key={role} value={role}>
+                    {role}
+                  </Option>
+                ))}
+              </BACCombobox>
+            </div>
+
+            <div className="requestFieldContainer">
+              <Label
+                id="userId"
+                weight="semibold"
+                className="fieldLabel"
+                required
+              >
+                <ContactIcon className="fieldIcon" />
+                User
+              </Label>
+              <Controller
+                name="user"
+                control={myForm.control}
+                rules={{
+                  required: "User is required",
+                }}
+                render={({ field }) => (
+                  <PeoplePicker
+                    ariaLabel="User"
+                    aria-describedby="userErr"
+                    aria-labelledby="userId"
+                    aria-invalid={
+                      myForm.formState.errors.user ? "true" : "false"
+                    }
+                    selectedItems={field.value ?? []}
+                    updatePeople={(items) => {
+                      if (items?.[0]?.Title) {
+                        field.onChange(items[0]);
+                      } else {
+                        field.onChange([]);
+                      }
+                    }}
+                  />
+                )}
+              />
+              {myForm.formState.errors.user && (
+                <Text role="alert" id="userErr" className="fieldErrorText">
+                  {myForm.formState.errors.user.message}
+                </Text>
+              )}
+            </div>
+          </form>
+        </FormProvider>
+      </DrawerBody>
+      <DrawerFooter>
+        {!addRole.isSuccess && (
+          <Button
+            style={{ marginLeft: "auto" }}
+            appearance="primary"
+            form="addRoleForm"
+            type="submit"
+            value="submit"
+            disabled={addRole.isLoading || addRole.isSuccess}
```

Comment:
No biggie, but addRole.isSuccess can never be true at this point, since it won't show this component if it is.

```suggestion
            disabled={addRole.isLoading}
```